pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -200,7 +155,75 @@ func (r *PvcCloneReconciler) cleanup(syncState *dvSyncState) error {
 		return nil
 	}
 
-	return r.reconcileVolumeCloneSourceCR(syncState)
+	r.log.V(3).Info("Cleanup initiated in dv pvc clone controller")
+
+	if err := r.reconcileVolumeCloneSourceCR(syncState); err != nil {
+		return err
+	}
+
+	if err := r.cleanupSizeDetectionPod(dv, syncState.pvc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *PvcCloneReconciler) cleanupSizeDetectionPod(dv *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) error {
+	if pvc != nil && !cc.ShouldDeletePod(pvc) {
+		return nil
+	}
+
+	nn := types.NamespacedName{Namespace: dv.Spec.Source.PVC.Namespace, Name: dv.Spec.Source.PVC.Name}
+	sourcePvc := &corev1.PersistentVolumeClaim{}
+	if err := r.client.Get(context.TODO(), nn, sourcePvc); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
```

Comment:
There are several places here you can return IgnoreNotFound() instead, but since it's a backport, let's do it in a follow-up PR.